src/v1/parser.rs

```diff
@@ -0,0 +1,814 @@
+        };
+        let block_size = if let Some(block_size_attr) = parts.next() {
+            let mut block_size_kv = block_size_attr.splitn(2, '=');
+            match block_size_kv.next() {
+                None => {
+                    return Err(ParseRowError(format!("Missing block_size")));
+                },
+                Some(k) if k != "block_size" => {
+                    return Err(ParseRowError(
+                        format!("Expected block_size attribute")));
+                },
+                Some(_) => {
+                    match block_size_kv.next() {
+                        Some(v) => u64::from_str_radix(v, 10)
+                            .map_err(|e| ParseRowError(
+                                format!("Cannot parse block size {:?}: {}", v, e)))?,
+                        None => return Err(ParseRowError(
+                            format!("Missing block size"))),
+                    }
+                },
+            }
+        } else {
+            return Err(ParseRowError(
+                format!("Missing block size attribute")));
+        };
+        Ok(Header {
+            version: version.to_string(),
+            hash_type: hash_type,
+            block_size: block_size,
+        })
+    }
+
+    /// Returns version of the signature file
+    pub fn get_version(&self) -> &str {
+        &self.version
+    }
+
+    /// Returns hashing algorithm
+    pub fn get_hash_type(&self) -> HashType {
+        self.hash_type
+    }
+
+    /// Returns block size
+    pub fn get_block_size(&self) -> u64 {
+        self.block_size
+    }
+}
+
+#[derive(Debug)]
+pub struct Footer(Vec<u8>);
+
+impl Footer {
+    fn parse(row: &[u8], hash_type: HashType)
+        -> Result<Footer, ParseRowError>
+    {
+        let digest_len = hash_type.get_size();
+        let hash_len = digest_len * 2;
+        if row.len() != hash_len {
+            return Err(ParseRowError(
+                format!("Footer length is {}, expected {}",
+                    row.len(), hash_len)));
+        }
+        let mut data = Vec::with_capacity(digest_len);
+        parse_hash(row, hash_len, &mut data)?;
+        Ok(Footer(data))
+    }
+}
+
+/// Entry hashes iterator
+#[derive(Debug, PartialEq)]
+pub struct Hashes {
+    data: Vec<u8>,
+    hash_type: HashType,
+}
+
+impl Hashes {
+    fn parse(row: &[u8], hash_type: HashType)
+        -> Result<Hashes, ParseRowError>
+    {
+        let digest_len = hash_type.get_size();
+        let hash_len = digest_len * 2;
+        let hashes_num = (row.len() + 1) / (hash_len + 1);
+        if row.len() != 0 && (row.len() + 1) % (hash_len + 1) != 0 {
+            return Err(ParseRowError(
+                format!("Hashes string length is {}, one hash should have exactly {} characters",
+                    row.len(), hash_len)));
+        }
+        let mut data = Vec::with_capacity(hashes_num * digest_len);
+        for hash in row.chunks(hash_len + 1) {
```

Comment:
I'm not sure we always require single space everywhere. Usually, people expect text files to ignore subsequent spaces.
So early `split_whitespace()` (as mentioned in another comment) should work better.